src/Avalonia.Controls/Presenters/ContentPresenter.cs

```diff
@@ -277,9 +277,23 @@ public void UpdateChild()
 
                 Child = newChild;
 
+                if (oldChild?.LogicalParent == this)
```

Comment:
`Parent` and `LogicalParent` are the same property, but because we use `newChild.Parent` below, it would probably best to use `oldChild?.Parent` here just to make it clear that it's the same thing to casual readers.